tools/devtool

```diff
@@ -731,6 +734,84 @@ cmd_tag() {
     say "Tag v$version created."
 }
 
+# Check if able to run firecracker.
+# https://github.com/firecracker-microvm/firecracker/blob/master/docs/getting-started.md#prerequisites
+# https://github.com/firecracker-microvm/firecracker/blob/master/docs/prod-host-setup.md
+ensure_kvm_rw () {
+    [[ -c /dev/kvm && -w /dev/kvm && -r /dev/kvm ]] || \
+        say_err "FAILED: user $(whoami) doesn't have permission to" \
+                "access /dev/kvm. You won't be able to run Firecracker"
+}
+check_SMT () {
+    (grep -q "^forceoff$\|^notsupported$" \
+	  /sys/devices/system/cpu/smt/control) ||
+	say_err "FAILED: Hyperthreading ENABLED."
+}
+check_KPTI () {
+    (grep -q "^Mitigation: PTI$" \
+	  /sys/devices/system/cpu/vulnerabilities/meltdown) || \
+	say_err "FAILED: KPTI NOT SUPPORTED"
+}
+check_KSM () {
+    (grep -q "^0$" /sys/kernel/mm/ksm/run) || \
+	say_err "FAILED: KSM ENABLED"
+}
+check_IBPB_IBRS () {
+    (grep -q "^Mitigation: Full generic retpoline, IBPB, IBRS_FW$"\
+	  /sys/devices/system/cpu/vulnerabilities/spectre_v2) || \
+	say_err "FAILED: retpoline, IBPB, IBRS: DISABLED."
+}
+check_L1TF () {
+    declare -a CONDITIONS=("Mitigation: PTE Inversion" "VMX: cache flushes")
+    for cond in "${CONDITIONS[@]}";
+    do (grep -q "$cond" /sys/devices/system/cpu/vulnerabilities/l1tf) ||
+	   say_err "FAILED: $cond: DISABLED";
+    done
+}
+check_swap () {
+    (grep -q "swap.img" /proc/swaps ) && \
+	say_err "FAILED: SWAP enabled"
+}
+check_kernver () {
+    KERN_MAJOR=4
+    KERN_MINOR=14
+    (uname -r | awk -v MAJOR=$KERN_MAJOR -v MINOR=$KERN_MINOR '{ split($0,kver,".");
+    if( (kver[1] + (kver[2] / 100) ) >=  MAJOR + (MINOR/100) )
```

Comment:
style nit: can you move the `say_err` inside the arithmetic `if` block without the `exit 0 / exit 1` magic?

Reply:
Thanks for checking this out.
I could remove the else condition, if we want to use say_err inside awk, I will need to export that function then calling through the system function call in awk, that will add more complexity to the function. I'll change it so it's shorter.